atc/api/resourceserver/versionserver/list.go

```diff
@@ -88,6 +90,11 @@ func (s *Server) ListResourceVersions(pipeline db.Pipeline) http.Handler {
 
 		w.WriteHeader(http.StatusOK)
 
+		acc := accessor.GetAccessor(r)
+		hideMetadata := !resource.Public() && !acc.IsAuthenticated()
```

Comment:
`acc.IsAuthenticated()` isn't enough - I need to also be a member of the team. This just checks that I'm logged in at all.

Reply:
I think the authorization check happens in the auth handler:

https://github.com/concourse/concourse/blob/03b05e7fe33033cfcaa0a41aff634293943ac8e0/atc/wrappa/api_auth_wrappa.go#L94-L95

At the this point, the user is either not logged in or logged in to the correct team, and we only want to hide metadata if the resource is private and the user is not logged in. We didn't like we had to get the accessor here to check if the user is logged in, is there a better way to get the information?

Reply:
I could also be logged in and able to view the pipeline because it's exposed - in that case the metadata should be hidden. I think you can just use `IsAuthorized`: https://github.com/concourse/concourse/blob/03b05e7fe33033cfcaa0a41aff634293943ac8e0/atc/api/accessor/accessor.go#L13

Reply:
Ah, that makes sense. Pushed a fix.